### PR TITLE
Load Oj instead of JSON.

### PR DESCRIPTION
**Pull request by jondeandres**

### CODE REVIEW

File: lib/rollbar.rb [hunk 1]
```diff
@@ -796,14 +796,7 @@ def require_core_extensions
     end
 
     def monkey_patch_socket?
-      return false unless defined?(ActiveSupport::VERSION::STRING)
```
Comment (jondeandres) [inline lib/rollbar.rb]: Testing failed with active support 4.2. So just monkey patch always active support is loaded.

**Comment by PetrKaleta** [comment]
You can't use Oj omg as its C extension gem! So what about JRuby omg

File: lib/rollbar.rb (hunk 1 above)
Comment (PetrKaleta) [inline lib/rollbar.rb]: I am just curious how this change is related with this issue

Comment (jondeandres) [inline lib/rollbar.rb]: @PetrKaleta ActiveSupport has some bugs when calling `#to_json` or `#as_json`. Most of the JSON serialization libraries calls somewhere `#as_json` if it's defined.

We've experienced many customers that in their payloads, for some reason, have objects that internally have an instance of `BasicSocket`. ActiveSupport then arrives to an infinite loop trying to execute `#as_json`. This caused by the monkey patch they add to `Object`.

And in this PR we realized we were having problems with all the ActiveSupport versions.

Comment (PetrKaleta) [inline lib/rollbar.rb]: Thanks for explanation, makes sense

**Pull request merged**